Update ThodaCSS and use updates

polybuildr opened the pull request.

polybuildr committed:
Update ThodaCSS to 38987ce

- Bug fixes involving margins, line-heights
- text-error and text-success classes
- Disabled form elements now look disabled

nisargjhaveri commented on src/app/views/sap/create_mission.php:
```diff
@@ -14,13 +14,12 @@
-                    echo "<p style=\"color:red\">Something went wrong. :/</p>";
-                }
-                ?>
+                    foreach ($errors as $error) { ?>
```
Are we using brackets? or the other syntax with `foreach () :`, `endforeach;`? Second one is much more readable in most cases when using across php tags.

polybuildr replied:
I agree. However, when it's in between another block of PHP, that makes things weird. @hharchani?

nisargjhaveri replied:
I just saw that.. The php tags should also be indented. In this case you've ignored that.

nisargjhaveri replied:
And doing all of this might make things messy.

hharchani replied:
I like the colon syntax more. It's more clear

polybuildr replied:
> I just saw that.. The php tags should also be indented. In this case you've ignored that.

You mean `<?php` should be indented one level? Or the code inside that should be indented one level?

> And doing all of this might make things messy.

That's exactly why I used the earlier `echo` syntax. It actually made things cleaner.

> I like the colon syntax more. It's more clear

Again, I agree. But even within another block of PHP?

nisargjhaveri replied:
I was talking about a pair of `<?php` `?>`. They should always be in the same level in new line, if multiline code block.

nisargjhaveri replied:
In this case, I certainly feel that echo-ing is much cleaner.

hharchani commented:
I like the colon syntax more. It's more clear

polybuildr committed:
Clean up success and error output message HTML

- Use ThodaCSS's text-error and text-success classes
- Remove .error and .success classes from sap_register.css

polybuildr commented:
Updated the PR. Consensus around `echo` in PHP is still not clear, so I've kept that part of the code as-is. ThodaCSS has been updated, and the `text-error` and `text-success` classes have been used.

nisargjhaveri commented on src/app/views/sap/create_mission.php:
```diff
@@ -15,10 +15,10 @@
                 if (isset($errors) && count($errors) !== 0) {
                     foreach ($errors as $error) {
-                        echo "<p style=\"color:red\">$error</p>";
+                        echo "<p class=\"text-error\">$error</p>";
```
Why escape quotes? :/

Please use different quotes, it does not hurt the eyes as this one do. :P

This is one of the reasons to not use echo, btw :P

polybuildr replied:
This is another convention issue that I was going to bring up.
- The reason to use `"` in strings that get displayed to the user is that there is generally usage of variable insertion and `'` ASCII apostrophes in human readable text.
- The reason to use `'` in PHP code is that `"` come up in other kinds of code, such as HTML, and hence need to be escaped.

Wherever I've escaped the quotes, I've gone with picking the first one over the second one. I'll change it to whatever convention we agree upon, but please let it slide for now.

nisargjhaveri replied:
I would personally prefer a string concatenation over escaping quotes! But, okay. No problem with this.

polybuildr replied:
That's actually the fun part. :P For some reason I hate string concatenation when variable insertion is possible. We'll discuss this later though. Not import right now, plus, it's not being introduced in this PR.

nisargjhaveri commented on src/app/views/sap/register.php:
```diff
@@ -63,13 +63,13 @@
             } else if (isset($success)) {
                 if ($success) {
-                    echo "<p class=\"success\">You have applied successfully. We shall review your application and get back to you.</p>";
+                    echo "<p class=\"text-success\">You have applied successfully. We shall review your application and get back to you.</p>";
```
The colour will change from darkgreen to green. You sure about this?

polybuildr replied:
Nice observation. :D Impressed by your reviewing. :P But yeah, I'm aware of this, and yeah, sure about this. :) They're not different enough to cause any issues in the application.

nisargjhaveri commented:
Looks fine. Merging.

nisargjhaveri commented:
I can't merge a closed PR! :P

polybuildr commented:
Wow. I got so many mails I wondered if you were having a stroke or there was some serious bug in GitHub or something. :P

nisargjhaveri commented:
I accidentally closed PR instead of merging. So, ...

polybuildr commented:
Yeah, I came here and then I got it. :P

polybuildr commented:
The number of mails Harshit must have got though! :P

nisargjhaveri commented:
One more! :P

Actually, not only Harshit. 3 more people are watching this repo!
